os/kernel/group/group_leave.c

```diff
@@ -414,26 +387,44 @@ void group_leave(FAR struct tcb_s *tcb)
 		/* Have all of the members left the group? */
 
 		if (group->tg_nmembers == 0) {
-#ifdef CONFIG_APP_BINARY_SEPARATION
-			if (IS_LOADED_MODULE(group)) {
+#ifdef CONFIG_BINFMT_LOADABLE
+			if (IS_BINARY_MAINTASK(tcb)) {
 				/* The region of stack is allocated in user RAM partition.
 				 * If it is loaded, whole RAM partition for binary is freed by unloading in group_release.
-				 * So the stack doesn't need to freed after unloading.
+				 * So the stack doesn't need to be freed after unloading.
 				 */
 
 				tcb->stack_alloc_ptr = NULL;
+#ifdef CONFIG_BINARY_MANAGER
+				/* Clear binary data and update binary state before group_release
+				 * because a group has a binary index to access binary table.
+				 */
+
+				binary_manager_clear_bindata(group->tg_binidx);
+#endif /* CONFIG_BINARY_MANAGER */
 			}
-#endif
-			/* Yes.. Release all of the resource held by the task group */
+#endif /* CONFIG_BINFMT_LOADABLE */
+
```

Comment:
Could you remove this extra line or the one at 409 please?
Seems like too many line spaces.